src/java/org/jivesoftware/openfire/group/GroupManager.java

```diff
@@ -213,11 +207,20 @@ public void adminRemoved(Group group, Map params) {
                 
                 // Remove only the collection of groups the member belongs to.
                 String member = (String) params.get("admin");
-                if(member != null) {
-	                groupMetaCache.remove(member);
-                }
+                evictCachedUserForGroup(member);
             }
 
+            private void evictCachedUserForGroup(String user) {
+                if(user != null) {
+                    JID userJid = new JID(user);
+                    if (XMPPServer.getInstance().isLocal(userJid)) {
+                        String username = userJid.getNode();
+                        synchronized (username.intern()) {

```

Comment:
I'm not crazy about synchronizing on an interned string. Was the lack of a lock a problem before?